### DISCUSSION TITLE
Listen to change event to update static site

### DISCUSSION THREAD

**Discussion by vincentole** [giscus/giscus#476] [Q&A] [answered]
Hello everyone,

first of all thanks to the people working on giscus for this fantastic tool! Its really amazing :) 

I would like to call an api route that updates my static page whenever a change (post, reaction) happened on giscus. I looked into the event messaging for this, but couldn't figure it out.

**Is it possible to explicitly listen for changes in giscus?** 

PS: I'm using Next.js and want to try On-Demand Revalidation

Best regards,
vincentole

**Comment by laymonage** [comment, marked as answer]
Hey, thanks for the feedback.

There is no such functionality on giscus, as it doesn't keep track of your discussions and it only loads your discussions on-demand by calling GitHub's API.

However, GitHub does provide [a webhook for Discussions](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#discussion) that I think could be useful for your use case.

**Reply by vincentole** [reply]
Hello, thanks for the information. 

I will have a look into webhooks then :)
